tracer/build/_build/Build.Shared.Steps.cs

```diff
@@ -103,15 +120,30 @@ partial class Build
         .After(CompileNativeLoader)
         .Executes(() =>
         {
-            // Copy native loader assets
-            var (arch, ext) = GetUnixArchitectureAndExtension();
-            var source = NativeLoaderProject.Directory / "bin" / "loader.conf";
-            var dest = MonitoringHomeDirectory / arch;
-            CopyFileToDirectory(source, dest, FileExistsPolicy.Overwrite);
+            if (IsLinux)
+            {
+                // Copy native loader assets
+                var (arch, ext) = GetUnixArchitectureAndExtension();
+                var source = NativeLoaderProject.Directory / "bin" / "loader.conf";
+                var dest = MonitoringHomeDirectory / arch;
+                CopyFileToDirectory(source, dest, FileExistsPolicy.Overwrite);
 
-            source = NativeLoaderProject.Directory / "bin" /
-                         $"{NativeLoaderProject.Name}.{ext}";
-            dest = MonitoringHomeDirectory / arch;
-            CopyFileToDirectory(source, dest, FileExistsPolicy.Overwrite);
+                source = NativeLoaderProject.Directory / "bin" / $"{NativeLoaderProject.Name}.{ext}";
+                dest = MonitoringHomeDirectory / arch;
+                CopyFileToDirectory(source, dest, FileExistsPolicy.Overwrite);
+            }
+            else if (IsOsx)
+            {
+                foreach (var (arch, folder, extension) in GetMacOsArchitectureAndExtension())
+                {
+                    var source = NativeLoaderProject.Directory / "bin" / "loader.conf";
+                    var dest = MonitoringHomeDirectory / folder;
+                    CopyFileToDirectory(source, dest, FileExistsPolicy.Overwrite);
+                    
+                    CopyFile(NativeLoaderProject.Directory / "bin" / $"{NativeLoaderProject.Name}.{arch}.{extension}",
+                             MonitoringHomeDirectory / folder / $"{NativeLoaderProject.Name}.{extension}",
+                             FileExistsPolicy.Overwrite);
+                }
+            }
         });
```

Comment:
No point in having a single `PublishNativeLoaderUnix` target if it has `if`/`else` inside. I suggest splitting it into two separate targets `PublishNativeLoaderLinux`, PublishNativeLoaderOsx`, and set `OnlyWhenStatic()` as appropriate